web/src/helpers/storage/search.ts

```diff
@@ -25,22 +25,39 @@ import { model } from "~/types/storage";
 import { copyApiModel } from "~/helpers/storage/api-model";
 import { buildModel } from "~/helpers/storage/model";
 
-function buildDrive(apiModel: apiModel.Config, name: string): model.Drive | undefined {
+function buildModelDevice(
+  apiModel: apiModel.Config,
+  name: string,
+  list: string,
+): model.Drive | undefined {
   const model = buildModel(apiModel);
-  return model.drives.find((d) => d.name === name);
+  return model[list].find((d) => d.name === name);
 }
 
 function deleteIfUnused(apiModel: apiModel.Config, name: string): apiModel.Config {
   apiModel = copyApiModel(apiModel);
 
-  const index = (apiModel.drives || []).findIndex((d) => d.name === name);
-  if (index === -1) return apiModel;
+  let list = "drives";
+  let index = (apiModel.drives || []).findIndex((d) => d.name === name);
+  if (index === -1) {
+    list = "mdRaids";
+    index = (apiModel.mdRaids || []).findIndex((d) => d.name === name);
+    if (index === -1) return apiModel;
+  }
 
-  const drive = buildDrive(apiModel, name);
-  if (!drive || drive.isUsed) return apiModel;
+  const device = buildModelDevice(apiModel, name, list);
+  if (!device || device.isUsed) return apiModel;
 
-  apiModel.drives.splice(index, 1);
+  apiModel[list].splice(index, 1);
   return apiModel;
 }
 
-export { deleteIfUnused };
+function addSearched(apiModel: apiModel.Config, data: data.SearchedDevice): apiModel.Config {
```

Comment:
I am not sure about the *search* concept and the usage of `list` parameter to distinguish drives and md raids. I feel the `list` concept is exposing something internal about the mode. I would prefer to have explicit hooks/helpers for drives and mds. For example:

* */helpers/storage/drive*:
  * `addDrive()`
* */helpers/storage/md-raid*:
  * `addMdRaid()`  
* */hooks/storage/drive*:
  * `useAddDrive()`
* */hooks/storage/md-raid*:
  * `useAddMdRaid()` 
  
I know that *drive* and *mdRaid* are basically the same at this moment, but they will differ at some point. I would share methods (e.g., `buildModelDevice`) to avoid repetition of common parts.